spec/components/candidate_interface/decoupled_references_review_component_spec.rb

```diff
@@ -1,5 +1,128 @@
+    end
+  end
+
+  context 'when editable is false' do
+    let(:reference) { create(:reference, :unsubmitted) }
+    let(:result) { render_inline(described_class.new(references: [reference], editable: false)) }
+
+    it 'fields cannot be changed' do
+      expect(result.css('.app-summary-card__body').text).not_to include 'Change'
+    end
+
+    it 'the reference cannot be deleted' do
+      expect(result.css('.app-summary-card__header').text).not_to include 'Delete referee'
+    end
+  end
+
+  context 'when reference state is "feedback_requested"' do
+    let(:feedback_requested) { create(:reference, :requested) }
+    let(:feedback_refused) { create(:reference, :refused) }
+
+    it 'a cancel link is available' do
+      result = render_inline(described_class.new(references: [feedback_requested, feedback_refused]))
+
+      feedback_requested_summary = result.css('.app-summary-card')[0]
+      feedback_refused_summary = result.css('.app-summary-card')[1]
+      expect(feedback_requested_summary.text).to include 'Cancel request'
+      expect(feedback_refused_summary.text).not_to include 'Cancel request'
+    end
+  end
+
+private
+
+  def status_table
+    af = create(:application_form)
+
+    not_requested_yet = create(:reference, :unsubmitted, application_form: af)
+    feedback_refused = create(:reference, :refused, application_form: af)
+    email_bounced = create(:reference, :email_bounced, application_form: af)
+    cancelled_at_end_of_cycle = create(:reference, :cancelled_at_end_of_cycle, application_form: af)
+    cancelled = create(:reference, :cancelled, application_form: af)
+    feedback_overdue = create(:reference, :feedback_overdue, application_form: af)
+    sent_less_than_5_days_ago = create(:reference, :sent_less_than_5_days_ago, application_form: af)
+    sent_more_than_5_days_ago = create(:reference, :sent_more_than_5_days_ago, application_form: af)
+    feedback_provided = create(:reference, :complete, application_form: af)
+
+    [
+      Status.new(not_requested_yet, :grey, 'not_requested_yet', 'not_requested_yet'),
+      Status.new(feedback_refused, :red, 'feedback_refused', 'declined'),
+      Status.new(email_bounced, :red, 'email_bounced', ''),
+      Status.new(cancelled_at_end_of_cycle, :orange, 'cancelled_at_end_of_cycle', 'cancelled_at_end_of_cycle'),
+      Status.new(cancelled, :orange, 'cancelled', 'cancelled'),
+      Status.new(feedback_overdue, :yellow, 'feedback_overdue', 'feedback_overdue'),
+      Status.new(sent_less_than_5_days_ago, :purple, 'feedback_requested', 'awaiting_reference_sent_less_than_5_days_ago'),
+      Status.new(sent_more_than_5_days_ago, :purple, 'feedback_requested', 'awaiting_reference_sent_more_than_5_days_ago'),
+      Status.new(feedback_provided, :green, 'feedback_provided', ''),
+    ]
+  end
+  # rubocop:disable RSpec/LeakyConstantDeclaration
+  Status = Struct.new(:reference, :colour, :status_identifier, :info_identifier)
+  # rubocop:enable RSpec/LeakyConstantDeclaration
```

Comment:
What's this?

Reply:
+1

Reply:
Rubocop wants you to use `stub_const('Status', Struct.new(:reference, :colour, :status_identifier, :info_identifier))` which is far less clear.

Reply:
Yeah, agree the assignment form is clearer. I've pushed a change which is a reasonable balance between the two and allows for removing the rule disabling.